packages/go_router/lib/src/go_router_delegate.dart

```diff
@@ -342,7 +342,9 @@ class GoRouterDelegate extends RouterDelegate<Uri>
         for (final match in matches) {
           // merge new params to keep params from previously matched paths, e.g.
           // /family/:fid/person/:pid provides fid and pid to person/:pid
-          params = {...params, ...match.decodedParams};
+          params = {...params, ...match.encodedParams};
+          // Add all the params to the next match
+          match.encodedParams.addAll(params);
```

Comment:
This can be simplified as
```dart
        var previouslyMatchedParams = <String, String>{};
        for (final match in matches) {
          assert(!previouslyMatchedParams.keys.any(match.encodedParams.containsKey), 'Duplicated parameter names');
          match.encodedParams.addAll(previouslyMatchedParams);
          previouslyMatchedParams = match.encodedParams;
        }
```

Reply:
done